### PR TITLE
Update docker build context to specific data needed.

### PR DESCRIPTION
**Pull request by devinrsmith**
Additionally, restructure deephaven image to include application and property file in default configuration directory.

### CODE REVIEW

File: docker-compose.yaml [hunk 1]
```diff
@@ -31,11 +31,9 @@ services:
   deephaven:
     container_name: deephaven
     build:
-      context: .
-      dockerfile: build/Dockerfile.deephaven
+      context: deephaven
```
Comment (kzk2000) [inline docker-compose.yaml:34]: @devinrsmith asking to learn
```
      dockerfile: build/Dockerfile.deephaven
      vs
      context: deephaven
```
is the latter the general convention for core developers?

My intend was to keep all `build` stuff in one place, not married to it, just felt natural. your refactoring seems to be more focused on compartmentalizing everything into sub units, was that your main intend here?

File: docker-compose.yaml [hunk 2]
```diff
@@ -31,11 +31,9 @@ services:
...
     environment:
-      JAVA_OPTS: -Xmx24g -Dio.deephaven.app.GcApplication.enabled=true -Ddeephaven.application.dir=/data/storage/app.d
```
Comment (kzk2000) [inline docker-compose.yaml:37]: is` -Ddeephaven.application.dir=/data/storage/app.d` deprecated, or is your proposed changed to move it to deephaven.prop the recommended/preferred way?

Comment (kzk2000) [reply docker-compose.yaml:37]: @devinrsmith or does it automatically default to app.d/cryptofeed.app as soon as an .app file exists in the container?

Comment (kzk2000) [reply docker-compose.yaml:37]: @devinrsmith can be a separate PR but we should update the `README.md` -> Project Structure after this as well.

**Comment by kzk2000** [comment]
@devinrsmith : I ended up merging it after reading up on the DH docs, makes all sense, thanks for contributing!

File: docker-compose.yaml (hunk 1 above)
Comment (devinrsmith) [reply docker-compose.yaml:34]: By default, `<context>/Dockerfile` will be used as the dockerfile. One of the docker best practices is to explicitly scope the context to only the files that are necessary to build the Dockerfile - this aids in docker build cache-ability. For example, when you set `context: .`, then whenever anything in the unrelated `./data` folder changes, a docker rebuild will not be able to use its previous cache.

File: docker-compose.yaml (hunk 2 above)
Comment (devinrsmith) [reply docker-compose.yaml:37]: Explicitly setting `deephaven.application.dir` is fine; but I think putting it under Deephaven's `<configDir>/app.d/` is how I would recommend most users set it up.

Comment (kzk2000) [reply docker-compose.yaml:37]: Agreed, I even moved the init_app.py there as well

**Pull request merged**